File: DeepSpaceJava/src/SuppliesPackage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author laura
 */
public class SuppliesPackage {
    
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    
    public SuppliesPackage(float power, float fuel, float shield){
        ammoPower = power;
        fuelUnits = fuel;
        shieldPower = shield;
    }
    
    public SuppliesPackage(SuppliesPackage s){
        copy=new SuppliesPackage(s.getAmmoPower(), s.getFuelUnits(), s.getShieldPower());
        return copy;  
    }
    
// Consultores
    
    public getAmmoPower(){
        return ammoPower;
    }
    
    public getFuelUnits(){
        return FuelUnits;
    }
    
    public getShieldPower(){
        return shieldPower;
    }
}

File: DeepSpaceRuby/lib/SuppliesPackage.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Deepspace
  class SuppliesPackage
    
    attr_reader:ammoPower

    attr_reader:fuelUnits

    attr_reader:shieldPower
    
    
    def initialize(power, fuel, shield)
      @ammoPower = power
      @fuelUnits = fuel
      @shieldPower = shield

    end

    def self.newCopy(s)
      new(s.ammoPower, s.fuelUnits, s.shieldPower)
    end
    
    def to_s
   
       "AmmoPower:  #{@ammoPower}\nFuelUnits:#{@fuelUnits} \nShieldPower: #{@shieldPower}"
    end
    
  end
  
end
